jnigen/lib/src/bindings/dart_generator.dart

```diff
@@ -589,7 +589,7 @@ class _$implClassName$typeParamsDef implements $implClassName$typeParamsCall {
         ? '($typeClassName).hashCode'
         : 'Object.hash($typeClassName, $hashCodeTypeClasses)';
     s.write('''
-class $typeClassName$typeParamsDef extends $_jType<$name$typeParamsCall> {
+final class $typeClassName$typeParamsDef extends $_jType<$name$typeParamsCall> {
```

Comment:
Nit: could have been a separate PR to ease reviewing. (I have already reviewed now, so keep it in here!)